Bump to v0.10

oacdutra opened the pull request:
This PR is an epic to update RGB to v0.10.

**Checklist**
- [x] Finish updating the dependencies
- [x] Add default schema interface (contract base)
- [x] Add RGB contract builder to RGB20 interface
- [x] Create a new endpoint to issue a new RGB20 contract
- [x] Create a new endpoint to generate an RGB invoice
- [x] Allow import RGB20 fungible contract
- [x] Update the PSBT creation operation
- [x] Create a new endpoint to generate a new consignment
- [x] Create a new endpoint to reveal blinded allocations
- [x] Create an official rgb21 interface (RGB-WG/rgb-wallet repo) *** 
- [x] Write a little RGB glossary explain the new terminologies

oacdutra committed:
update dependencies

oacdutra committed:
finish update dependencies

oacdutra committed:
move previous operations to rgb_legacy

oacdutra committed:
create issue contract operation

oacdutra committed:
fix typo

cryptoquick commented on Cargo.toml:
```diff
@@ -45,58 +49,61 @@ payjoin = { version = "0.6.0", features = ["sender"] }
 
 [target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-amplify = "3.13.0"
+amplify_legacy = { package = "amplify", version = "3.13.0" }
```
Is there any reason to do legacy here? We should have the same functionality either way, no?

oacdutra replied:
Yes, amplify v4 was released!

The major milestone is a full compatibility with wasm. 

My goal is remove all *_legacy crates after finish the work!

cryptoquick commented on Cargo.toml:
```diff
@@ -45,58 +49,61 @@ payjoin = { version = "0.6.0", features = ["sender"] }
 pretty_env_logger = "0.4.0"
 regex = "1.7.0"
 reqwest = { version = "0.11.13", features = ["json"] }
+rgb-std = { version = "0.10.0-rc.1" }
+rgb-wallet = {version = "0.10.0-beta.1" }
```
Looks like a few of these deps need to be updated already...
rgb-std is in rc.2 now, and rgb-wallet is in rc.1.

oacdutra replied:
Yes, Maxim launch final release today!

I will update ASAP.

cryptoquick commented on src/bin/bitmaskd.rs:
```diff
@@ -67,6 +85,7 @@ async fn main() -> Result<()> {
     pretty_env_logger::init();
 
     let app = Router::new()
+        .route("/issue-contract", post(issue_contract))
```
Please don't make this a new route, just replace the /issue route. It'd reduce maintenance complexity.

oacdutra replied:
I was thinking during the migration, creating routes with a "v2" prefix, so we will be able to remove the previous ones as it is finished. After completing everything, we remove that "v2" prefix.

What do you think?

cryptoquick replied:
No. We haven't even hit 1.0.

cryptoquick commented on src/data/structs.rs:
```diff
@@ -84,6 +84,24 @@ pub struct IssueRequest {
+    pub seal: String,
+    /// The name of the iface (ex: RGB20)
+    pub iface: String,
+}
```
Please just modify the existing struct for issue_asset.

cryptoquick reviewed (changes requested):
Just some course corrections if you don't mind.

oacdutra committed:
bump to official release of the rgb v0.10 and fix minor issues

oacdutra committed:
feat: create rgb invoice

oacdutra committed:
feat: create full rgb psbt

oacdutra committed:
feat: rgb transfer

oacdutra committed:
feat: accept transfer

oacdutra commented:
Hi @cryptoquick and @josediegorobles 

I implemented the main functions of RGB. Right now, I'm finishing validating the transfer. After that, I will be working on improvements and testability.

I'm pushing the code for you to follow the work.

cryptoquick commented on Cargo.toml:
```diff
@@ -11,6 +11,7 @@ edition = "2021"
 license = "MIT"
 homepage = "https://bitmask.app"
 repository = "https://github.com/diba-io/bitmask-core"
+rust-version = "1.60.0"
```
Why did you specify the rust-version, btw?

cryptoquick commented on src/lib.rs:
```diff
@@ -1,22 +1,66 @@
+use psbt::Psbt;
+use rgbstd::containers::BindleContent;
+use rgbstd::contract::ContractId;
+use rgbstd::contract::GraphSeal;
```
All this stuff is antipattern... I think it'd be best to move the things using these things and put them in their own modules. They shouldn't be in the top-level lib. In the very least, it should be scoped to just the rgb ops module.

cryptoquick commented on src/lib.rs:
```diff
@@ -196,73 +239,168 @@ pub fn list_assets(contract: &str) -> Result<Vec<AssetResponse>> {
-        id,
-        asset_id,
+        id: id.clone(),
+        asset_id: id,
```
Why are these both the same?

cryptoquick commented on tests/rgb_test_utis.rs:
```diff
@@ -0,0 +1,139 @@
+use std::{convert::Infallible, str::FromStr};
```
This file name is mispelled.

cryptoquick reviewed (changes requested):
I appreciate the terminology improvements.
I'll make a PR into this one for adding Carbonado persistence methods.
Aside from that, just a few more questions and comments.

josediegorobles reviewed (changes requested):
I'm seeing the code, good work. For now i can't add to the suggestions of @cryptoquick. One thing, there are lots of changes in the API, and that's necessary because the changes in rgb but that means that we will need a little documentation about what needs to be send and what is received from the methods in lib and web. As we are using structs for that, a little description of every field and an example ther could be very useful when the job must be done in the extension (and for other people that will be using bitmask-core)

oacdutra committed:
style: apply lint

oacdutra committed:
fix: create valid transfers

oacdutra committed:
fix: pay and accept transfer

oacdutra committed:
refactor: remove old version of the rgb

oacdutra committed:
refactor: organize rgb modules

oacdutra committed:
refactor: apply lint

oacdutra commented:
Hi @cryptoquick and @josediegorobles,

I finish the migration to the new version. 

**Some Notes**
- For all interactions between RGB and Blockchain, I implemented with esplora instead of electrum, because esplora had support for wasm32. 
- I needed to add the v0.30 version of rust-bitcoin because of the RGB. I tried to update the whole project for it, but it was getting too much work. So I'll leave that for another time.
- I updated **descriptor-wallet**, **bitcoin_blockchain**, **bitcoin_scripts** and **psbt** crates to using _amplify v4_. This update allows us to use BP-WG libs without custom implementations. After Maxim approves the branches, I will update them to official releases.

oacdutra committed:
feat: rgb stock endpoints

oacdutra committed:
doc: update docker instructions

cryptoquick committed:
Merge branch 'development' into CD/bump-0.10

cryptoquick committed:
Update crates.

cryptoquick committed:
Add info logs to pay_invoice_error_test... Seems flakey.

oacdutra committed:
fix: merge conflicts

cryptoquick committed:
Merge pull request #163 from diba-io/HT/bump-0.10-updates

Bump to v0.10 updates

oacdutra committed:
fix: wasm compilation and tests

oacdutra committed:
fix: merge conflicts

cryptoquick reviewed (approved):
Really great work so far! This deprecates so much code! I love it.

Next, I'll work on Carbonado support, and then you can use that to build the WASM asset transfer end-to-end test.

cryptoquick commented:
@josediegorobles I'm going to merge this and additional documentation can be added in later PRs.

josediegorobles commented:
> @josediegorobles I'm going to merge this and additional documentation can be added in later PRs.

It's ok but next time it's better to wait to have been reviewed by two a so big and with so much changes PR.
